tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/MetaSyntax.java

```diff
@@ -0,0 +1,52 @@
+package fi.helsinki.cs.tmc.langs.domain;
+
+public class MetaSyntax {
+    
+    private static final String BEGIN_SOLUTION = "BEGIN[ \\t]+SOLUTION";
+    private static final String END_SOLUTION = "END[ \\t]+SOLUTION";
+    private static final String SOLUTION_FILE = "SOLUTION[ \\t]+FILE";
+    private static final String SPACES = "[ \\t]*";
+    private static final String STUB = "STUB:[ \\t]*";
+   
+    private String commentStartRegex;
+    private String commentEndRegex;
+    private String beginSolutionRegex;
+    private String endSolutionRegex;
+    private String solutionFileRegex;
+    private String stubRegex;
+    private String stubMarker;
+
+    public MetaSyntax(String commentStartSyntax, String commentEndSyntax) {
+        this.commentStartRegex   = SPACES + commentStartSyntax + SPACES;
+        this.commentEndRegex     = SPACES + commentEndSyntax   + SPACES;
+        this.beginSolutionRegex  = commentStartRegex  + BEGIN_SOLUTION + commentEndRegex;
+        this.endSolutionRegex    = commentStartRegex  + END_SOLUTION   + commentEndRegex;
+        this.solutionFileRegex   = commentStartRegex  + SOLUTION_FILE  + commentEndRegex;
+        this.stubRegex           = commentStartRegex  + STUB + "(.*)"  + commentEndRegex;
+        this.stubMarker          = commentStartSyntax + STUB; // Initial spaces left outside stubMarker on purpose
+    }
+
+    public String getCommentEndRegex() {
+        return commentEndRegex;     // eg. " -->    "
```

Comment:
Make these line ending comments into minimalJavaDoc comments instead. It takes a bit more space but helps when using these methods from other classes.